src/module/system/action-macros/helpers.ts

```diff
@@ -63,10 +66,59 @@ export class ActionMacroHelpers {
         }
     }
 
-    static defaultCheckContext<ItemType extends ItemPF2e<ActorPF2e>>(
-        options: CheckContextOptions<ItemType>,
-        data: CheckContextData<ItemType>,
-    ): CheckContext<ItemType> | undefined {
+    static async resolveWieldedWeaponFromItemParameter(
+        options: Partial<CombatManeuverActionUseOptions>,
+        traits: WeaponTrait | WeaponTrait[],
+    ): Promise<Partial<CombatManeuverActionPassthroughOptions>> {
+        const original = options.item;
+        const passthrough: CombatManeuverActionPassthroughOptions = { ...options, item: undefined };
+        if (original && typeof original === "string") {
+            if (original.match(/^Actor\.[a-zA-Z0-9]+\.Item\.[a-zA-Z0-9]+$/)) {
+                const item = await fromUuid<ItemPF2e<ActorPF2e>>(original);
+                if (!item) {
+                    return Promise.reject(
```

Comment:
From my limited digging, I think you can just throw. In an async method it appears its the same either way.

Reply:
Seems to be working the same 👍